src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,165 @@
+	 * @param filePath is the path for the function file (.cs for C# functions)
+	 */
+	private refactorAzureFunction(filePath: string): void {
+		let defaultBindedFunctionText = fs.readFileSync(filePath, 'utf-8');
+		// Replace default binding text
+		let newValueLines = defaultBindedFunctionText.split(os.EOL);
+		const defaultLineSet = new Set(defaultSqlBindingTextLines);
+		let replacedValueLines = [];
+		for (let defaultLine of newValueLines) {
+			// Skipped lines
+			if (defaultLineSet.has(defaultLine.trimStart())) {
+				continue;
+			} else if (defaultLine.trimStart() === defaultBindingResult) { // Result change
+				replacedValueLines.push(defaultLine.replace(defaultBindingResult, sqlBindingResult));
+			} else {
+				// Normal lines to be included
+				replacedValueLines.push(defaultLine);
+			}
+		}
+		defaultBindedFunctionText = replacedValueLines.join(os.EOL);
+		fs.writeFileSync(filePath, defaultBindedFunctionText, 'utf-8');
+	}
+}
+
+// A C# Azure Functions project must be present in order to create a new Azure Function for the table
+async function isAzureFunctionProjectOpen(): Promise<boolean> {
+	if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) {
+		return false;
+	}
+	const projFile = await getProjectFile();
+	const hostFile = await getHostFile();
+	return projFile !== undefined && hostFile !== undefined;
+}
+
+// gets the azure functions project file path
+async function getProjectFile(): Promise<string | undefined> {
+	const projFiles = await vscode.workspace.findFiles('**/*.csproj');
+	return projFiles.length > 0 ? projFiles[0].fsPath : undefined;
+}
+
+// gets the host file config file path
+async function getHostFile(): Promise<string | undefined> {
+	const hostFiles = await vscode.workspace.findFiles('**/host.json');
+	return hostFiles.length > 0 ? hostFiles[0].fsPath : undefined;
+}
+
+// gets the local.settings.json file path
+async function getSettingsFile(): Promise<string | undefined> {
+	const settingsFiles = await vscode.workspace.findFiles('**/local.settings.json');
+	return settingsFiles.length > 0 ? settingsFiles[0].fsPath : undefined;
+}
+
+// retrieves the new function file once the file is created
+function getNewFunctionFile(): Promise<string> {
+	return new Promise((resolve) => {
+		const watcher = vscode.workspace.createFileSystemWatcher(new vscode.RelativePattern(
+			vscode.workspace.workspaceFolders[0], '**/*.cs'), false, true, true);
+		watcher.onDidCreate((e) => {
+			resolve(e.fsPath);
+			watcher.dispose();
+		});
+	});
+}
+
+// adds the required nuget package to the project
+async function addNugetReferenceToProjectFile(): Promise<void> {
+	// Make sure the nuget source is added
+	const currentSources = await executeCommand('dotnet nuget list source');
+	if (currentSources.indexOf(sqlBindingNugetSource) === -1) {
+		await executeCommand(`dotnet nuget add source ${sqlBindingNugetSource}`);
+	}
+	const projFile = await getProjectFile();
```

Comment:
Right now you're just grabbing the first project file that is returned from the findFiles call. If the user has multiple projects open and the first one isn't a azure functions project then this is going to be getting the incorrect one.

I believe STS already has logic to get the project based on the existence of the host.json at the same level - you should likely be doing something similar. (first check whether host.json is guaranteed to be at the same level. But that would be a better place to start for now since I imagine other locations aren't as common)

Reply:
In order to address this I added a new prompt: 
![image](https://user-images.githubusercontent.com/23587151/151630803-ef3df4f9-8a4f-41c9-923a-610b30a242c2.png)